units/commit_queue.go

```diff
@@ -66,6 +67,46 @@ func NewCommitQueueJob(env evergreen.Environment, queueID string, id string) amb
 	return job
 }
 
+func (j *commitQueueJob) TryUnstick(cq *commitqueue.CommitQueue) {
+	//unstuck the queue if the patch is done.
+	nextItem := cq.Next()
+	if nextItem != nil {
+		patchIdString := nextItem.Issue
+		if !patch.IsValidId(patchIdString) {
+			j.AddError(errors.Errorf("error unsticking queue. Patch id  '%s' is not an object id", nextItem.Issue))
+			return
+		}
+		patchDoc, err := patch.FindOne(patch.ById(patch.NewId(nextItem.Issue)).WithFields(patch.FinishTimeKey))
+		if err != nil {
+			j.AddError(errors.Wrapf(err, "error determining if patch is done for %s", j.QueueID))
+			return
+		}
+		if patchDoc == nil {
+			j.AddError(errors.Errorf("could not unstick queue. patch %s not found", patchIdString))
+			return
+		}
+
+		//patchisdone
+		patchFinishTime := patchDoc.FinishTime
+		if !util.IsZeroTime(patchFinishTime) {
+			j.dequeue(cq, nextItem)
+			status := evergreen.MergeTestSucceeded
+			event.LogCommitQueueConcludeTest(patchIdString, status)
+			grip.Info(message.Fields{
+				"source":             "commit queue",
+				"job_id":             j.ID(),
+				"item_id":            nextItem.Issue,
+				"project_id":         cq.ProjectID,
+				"processing_seconds": time.Since(cq.ProcessingUpdatedTime).Seconds(),
+				"timeSincePatchDone": time.Since(patchFinishTime).Seconds(),
```

Comment:
Sorry I have so many nits!!! By convention (I realize none of our conventions match) we use underscores and not camelcase for logging.